virtualizarr/utils.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional, Union
+
+# TODO: importing fsspec and s3fs to get typing. Is there a better way incase these are optional deps?
+from s3fs.core import S3File
+from fsspec.implementations.local import LocalFileOpener
+
+
+def _fsspec_openfile_from_filepath(*, filepath: str, reader_options: Optional[dict] = {'storage_options':{'key':'', 'secret':'', 'anon':True}}) -> Union[S3File, LocalFileOpener]:
+    """Utility function to facilitate reading remote file paths using fsspec.
+
+    :param filepath: Input filepath
+    :type filepath: str
+    :param reader_options: Dict containing options to pass to fsspec file reader. Default: {'storage_options':{'key':'', 'secret':'', 'anon':True}}
+    :type reader_options: Optional[dict]
+    :rtype: Union[S3File, LocalFileOpener]
```

Comment:
This is not the same docstring style we're using in the rest of the package!

Reply:
Good catch! Just to check, are we using the `numpy` style?

Reply:
Whatever style this is, which I think is numpy?

https://github.com/TomNicholas/VirtualiZarr/blob/3ba30c7c46c725e0e2c637a173445f7a403b06a7/virtualizarr/xarray.py#L38